### PR TITLE
V1alpha2 doc 15155

### PR DESCRIPTION
**Pull request by anoipm**
**Description**

Changes proposed in this pull request:

- updated documentation of function's CRD - adaptation to v1alpha2

**Related issue(s)**
See also #15155

### CODE REVIEW

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 1]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
 | **spec.runtimeImageOverride**                 |       No       | Specifies the runtimes image which must be used instead of the default one. |
-| **spec.type**                          |      No       | Defines that you use a Git repository as the source of Function's code and dependencies. It must be set to `git`. |
-| **spec.source**                          |      Yes       | Provides the Function's full source code or the name of the Git directory in which the code and dependencies are stored.     |
-| **spec.baseDir**                          |      No       | Specifies the relative path to the Git directory that contains the source code from which the Function will be built​. |
-| **spec.reference**                        |      No       | Specifies either the branch name or the commit revision from which the Function Controller automatically fetches the changes in Function's code and dependencies. |
+| **spec.source**                               |      Yes       | Defines Function's source code or Git repository with it. |
```
Comment (dbadura) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:118]: I would describe it:
Contains specification of function. Only one configuration is allowed

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 2]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.source.inline**                        |       No       | Specifies Function's source code and dependencies. Can't be used at the same time with **spec.source.gitRepository**. |
```
Comment (dbadura) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:119]: Configure function as inline.

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 3]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.source.inline.dependencies**           |      Yes       | Specifies the Function's dependencies. |
+| **spec.source.inline.source**                 |      Yes       | Provides the Function's full source code of Function. |
+| **spec.source.gitRepository**                 |       No       | Specifies object pointing to a Git repository with the Function's source code and dependencies. Can't be used at the same time with **spec.source.inline**. |
```
Comment (dbadura) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:122]: Configure function as git sourced.

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 4]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.source.gitRepository.auth.type**       |      Yes       | Defines if you must authenticate to the repository with a password or token (`basic`), or an SSH key (`key`). For SSH, this parameter must be set to `key`. |
+| **spec.source.gitRepository.auth.secretName** |      Yes       | Specifies the name of the Secret with credentials used by the Function Controller to authenticate to the Git repository in order to fetch the Function's source code and dependencies. This Secret must be stored in the same Namespace as the GitRepository CR. |
+| **spec.env**                             |       No       | Specifies environment variables you need to export for the Function. You can export them either directly in the Function CR's spec or define them in a [ConfigMap](../00-configuration-parameters/svls-02-environment-variables.md#define-environment-variables-in-a-config-map). |
+| **spec.resourceConfiguration**                |       No       | Specifies resources requested by Function and build Job. |
+| **spec.resourceConfiguration.function**       |      Yes       | Specifies resources requested by the Function's Pod. |
```
Comment (dbadura) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:131]: It's not required as far as I know

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 5]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.resourceConfiguration.function.resources**                       |       No       | Defines amount of resources available for the Function's Pod to use. Can't be used at the same time with **spec.resourceConfiguration.function.profile**. |
+| **spec.resourceConfiguration.function.resources.limits.cpu**            |       No       | Defines the maximum number of CPUs available for the Function's Pod to use. |
+| **spec.resourceConfiguration.function.resources.limits.memory**         |       No       | Defines the maximum amount of memory available for the Function's Pod to use. |
+| **spec.resourceConfiguration.function.resources.requests.cpu**          |       No       | Specifies the number of CPUs requested by the Function's Pod to operate. |
+| **spec.resourceConfiguration.function.resources.requests.memory**       |       No       | Specifies the amount of memory requested by the Function's Pod to operate. |
+| **spec.resourceConfiguration.build**          |      Yes       | Specifies resources requested by the build Job's Pod. |
```
Comment (dbadura) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:138]: It's not required as far as I know

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md (hunk 1 above)
Comment (anoipm) [reply docs/05-technical-reference/00-custom-resources/svls-01-function.md:118]: IMO whole CRD "Contains source specification of function" :)
I have no idea for good description of this aggregate field.

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md (hunk 2 above)
Comment (anoipm) [reply docs/05-technical-reference/00-custom-resources/svls-01-function.md:119]: changed

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md (hunk 3 above)
Comment (anoipm) [reply docs/05-technical-reference/00-custom-resources/svls-01-function.md:122]: changed

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md (hunk 4 above)
Comment (anoipm) [reply docs/05-technical-reference/00-custom-resources/svls-01-function.md:131]: fixed

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md (hunk 5 above)
Comment (anoipm) [reply docs/05-technical-reference/00-custom-resources/svls-01-function.md:138]: fixed

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 6]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
 | **spec.runtimeImageOverride**                 |       No       | Specifies the runtimes image which must be used instead of the default one. |
-| **spec.type**                          |      No       | Defines that you use a Git repository as the source of Function's code and dependencies. It must be set to `git`. |
-| **spec.source**                          |      Yes       | Provides the Function's full source code or the name of the Git directory in which the code and dependencies are stored.     |
-| **spec.baseDir**                          |      No       | Specifies the relative path to the Git directory that contains the source code from which the Function will be built​. |
-| **spec.reference**                        |      No       | Specifies either the branch name or the commit revision from which the Function Controller automatically fetches the changes in Function's code and dependencies. |
+| **spec.source**                               |      Yes       | Defines Function's source code or Git repository with it. Only one source configuration is allowed. |
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:118]:
```suggestion
| **spec.source**                               |      Yes       | Defines the Function's source code or Git repository. Only one source configuration is allowed. |
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 7]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.source.inline**                        |       No       | Configure function as inline. |
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:119]:
```suggestion
| **spec.source.inline**                        |       No       | Defines Function as the inline Function. |
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 8]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.source.inline.dependencies**           |       No       | Specifies the Function's dependencies. |
+| **spec.source.inline.source**                 |      Yes       | Provides the Function's full source code of Function. |
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:121]:
```suggestion
| **spec.source.inline.source**                 |      Yes       | Provides the Function's full source code. |
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 9]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.source.gitRepository**                 |       No       | Configure function as git sourced. |
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:122]:
```suggestion
| **spec.source.gitRepository**                 |       No       | Defines Function as git-sourced. |
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 10]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.source.gitRepository.url**             |      Yes       | Provides the address to the Git repository with the Function's code and dependencies. Depending on whether the repository is public or private and what authentication method is used to access it, the URL must start with the `http(s)`, `git`, or `ssh` prefix, and end with the `.git` suffix. |
+| **spec.source.gitRepository.baseDir**         |       No       | Specifies the relative path to the Git directory that contains the source code from which the Function will be built. |
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:124]:
```suggestion
| **spec.source.gitRepository.baseDir**         |       No       | Specifies the relative path to the Git directory that contains the source code from which the Function is built. |
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 11]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.source.gitRepository.reference**       |       No       | Specifies either the branch name or the commit revision from which the Function Controller automatically fetches the changes in Function's code and dependencies. |
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:125]:
```suggestion
| **spec.source.gitRepository.reference**       |       No       | Specifies either the branch name or the commit revision from which the Function Controller automatically fetches the changes in the Function's code and dependencies. |
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 12]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.resourceConfiguration.build.resources.limits.cpu**            |       No       | Defines the maximum number of CPUs available to use for the Kubernetes Job's Pod responsible for building the Function's image.      |
+| **spec.resourceConfiguration.build.resources.limits.memory**         |       No       | Defines the maximum amount of memory available for the Job's Pod to use.      |
+| **spec.resourceConfiguration.build.resources.requests.cpu**          |       No       | Specifies the number of CPUs requested by the build Job's Pod to operate.       |
+| **spec.resourceConfiguration.build.resources.requests.memory**       |       No       | Specifies the amount of memory requested by the build Job's Pod to operate.               |
+| **spec.replicas**                             |       No       | Defines exact number of Function's Pods to run at a time. Can't be used at the same time with **spec.scaleConfig**. |
+| **spec.scaleConfig**                          |       No       | Defines minimum and maximum number of Functions Pods to run at a time. Can't be used at the same time with **spec.replicas**. |
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:146]:
```suggestion
| **spec.scaleConfig**                          |       No       | Defines minimum and maximum number of Function's Pods to run at a time. Can't be used at the same time with **spec.replicas**. |
```

File: docs/05-technical-reference/00-custom-resources/svls-02-gitrepository.md [hunk 13]
```diff
@@ -2,49 +2,6 @@
 title: Git Repository
 ---
 
-The `gitrepositories.serverless.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format used to define and manage Git repositories that store the Function's source code and dependencies. To get the up-to-date CRD and show the output in the YAML format, run this command:
+>**WARNING:** Current **apiVersion** is `serverless.kyma-project.io/v1alpha2`. Version `serverless.kyma-project.io/v1alpha1` is still supported but depreciated - for documentation see [GitRepository for version 2.5](../../../2.5/05-technical-reference/00-custom-resources/svls-02-gitrepository.md)
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-02-gitrepository.md:5]:
```suggestion
>**WARNING:** The current API version is `serverless.kyma-project.io/v1alpha2`. The `serverless.kyma-project.io/v1alpha1` version is still supported but deprecated - For the v1alpha1 version, see the [previous GitRepository documentation](https://github.com/kyma-project/kyma/blob/release-2.5/docs/05-technical-reference/00-custom-resources/svls-02-gitrepository.md)
```

File: docs/05-technical-reference/00-custom-resources/svls-02-gitrepository.md [hunk 14]
```diff
@@ -2,49 +2,6 @@
...
-| Custom resource           | Description                   |
-| ------------------- | ------------------------------------------------------------------------------------------------------------ |
-| [Function](./svls-01-function.md)     | Stores the Function's source code and dependencies on a cluster.  |
-
-These components use this CR:
-
-| Component           | Description                              |
-| ------------------- | ------------------------------------------------------------------------------------------------------------ |
-| Function Controller | Uses the GitRepository CR to locate the Function's source code and dependencies in a Git repository.
+The `gitrepositories.serverless.kyma-project.io` CustomResourceDefinition (CRD) was removed and definitions was moved to the `functions.serverless.kyma-project.io` CRD - see [Function](../00-custom-resources/svls-01-function.md)
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-02-gitrepository.md:7]:
```suggestion
The `gitrepositories.serverless.kyma-project.io` CustomResourceDefinition (CRD) was removed and definitions were moved to the `functions.serverless.kyma-project.io` CRD. For more information, see [Function](../00-custom-resources/svls-01-function.md)
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md (hunk 1 above)
Comment (dbadura) [reply docs/05-technical-reference/00-custom-resources/svls-01-function.md]:
```
Contains the function specification. Only one specification is allowed.
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 15]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
-| **spec.buildResources.limits.memory**         |       No       | Defines the maximum amount of memory available for the Job's Pod to use.      |
-| **spec.buildResources.requests.cpu**          |       No       | Specifies the number of CPUs requested by the build Job's Pod to operate.       |
-| **spec.buildResources.requests.memory**       |       No       | Specifies the amount of memory requested by the build Job's Pod to operate.               |
-| **spec.runtime**                         |       No       | Specifies the runtime of the Function. The available values are `nodejs14`, `nodejs16`, and `python39`. It is set to `nodejs14` unless specified otherwise.  |
+| **metadata.labels**                          |       No       | Specifies the Function's Pod labels.    |
+| **spec.runtime**                         |      Yes       | Specifies the runtime of the Function. The available values are `nodejs14`, `nodejs16`, and `python39`. It is set to `nodejs14` unless specified otherwise.  |
```
Comment (dbadura) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:116]:
```suggestion
| **spec.runtime**                         |      Yes       | Specifies the runtime of the Function. The available values are `nodejs14`, `nodejs16`, and `python39`. |
```
in v1alpha2 we don't set default runtime

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md (hunk 1 above)
Comment (anoipm) [reply docs/05-technical-reference/00-custom-resources/svls-01-function.md]: done

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 16]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
 | **spec.runtimeImageOverride**                 |       No       | Specifies the runtimes image which must be used instead of the default one. |
-| **spec.type**                          |      No       | Defines that you use a Git repository as the source of Function's code and dependencies. It must be set to `git`. |
-| **spec.source**                          |      Yes       | Provides the Function's full source code or the name of the Git directory in which the code and dependencies are stored.     |
-| **spec.baseDir**                          |      No       | Specifies the relative path to the Git directory that contains the source code from which the Function will be built​. |
-| **spec.reference**                        |      No       | Specifies either the branch name or the commit revision from which the Function Controller automatically fetches the changes in Function's code and dependencies. |
+| **spec.source**                               |      Yes       | Contains the function specification. Only one specification is allowed. |
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:118]:
```suggestion
| **spec.source**                               |      Yes       | Contains the Function's specification. Only one specification is allowed. |
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 17]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
-| **spec.reference**                        |      No       | Specifies either the branch name or the commit revision from which the Function Controller automatically fetches the changes in Function's code and dependencies. |
+| **spec.source**                               |      Yes       | Contains the Function's specification. Only one specification is allowed. |
+| **spec.source.inline**                        |       No       | Defines Function as the inline Function. |
+| **spec.source.inline.dependencies**           |       No       | Specifies the Function's dependencies. |
+| **spec.source.inline.source**                 |      Yes       | Provides the Function's full source code. |
+| **spec.source.gitRepository**                 |       No       | Defines Function as git-sourced. |
+| **spec.source.gitRepository.url**             |      Yes       | Provides the address to the Git repository with the Function's code and dependencies. Depending on whether the repository is public or private and what authentication method is used to access it, the URL must start with the `http(s)`, `git`, or `ssh` prefix, and end with the `.git` suffix. |
```
Comment (dbadura) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:123]:
```suggestion
| **spec.source.gitRepository.url**             |      Yes       | Provides the address to the Git repository with the Function's code and dependencies. Depending on whether the repository is public or private and what authentication method is used to access it, the URL must start with the `http(s)`, `git`, or `ssh` prefix. |
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 18]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.source.gitRepository.baseDir**         |       No       | Specifies the relative path to the Git directory that contains the source code from which the Function is built. |
+| **spec.source.gitRepository.reference**       |       No       | Specifies either the branch name or the commit revision from which the Function Controller automatically fetches the changes in the Function's code and dependencies. |
```
Comment (dbadura) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:125]:
```suggestion
| **spec.source.gitRepository.reference**       |       No       | Specifies either the branch name, tag or the commit revision from which the Function Controller automatically fetches the changes in the Function's code and dependencies. |
```

File: docs/05-technical-reference/00-custom-resources/svls-01-function.md [hunk 19]
```diff
@@ -100,25 +112,40 @@ This table lists all the possible parameters of a given resource together with t
...
+| **spec.source.gitRepository.auth**            |       No       | Specifies that you must authenticate to the Git repository. Required for SSH. |
+| **spec.source.gitRepository.auth.type**       |      Yes       | Defines if you must authenticate to the repository with a password or token (`basic`), or an SSH key (`key`). For SSH, this parameter must be set to `key`. |
+| **spec.source.gitRepository.auth.secretName** |      Yes       | Specifies the name of the Secret with credentials used by the Function Controller to authenticate to the Git repository in order to fetch the Function's source code and dependencies. This Secret must be stored in the same Namespace as the GitRepository CR. |
```
Comment (dbadura) [inline docs/05-technical-reference/00-custom-resources/svls-01-function.md:128]:
```suggestion
| **spec.source.gitRepository.auth.secretName** |      Yes       | Specifies the name of the Secret with credentials used by the Function Controller to authenticate to the Git repository in order to fetch the Function's source code and dependencies. This Secret must be stored in the same Namespace as the Function CR. |
```

File: docs/05-technical-reference/00-custom-resources/svls-02-gitrepository.md [hunk 20]
```diff
@@ -2,49 +2,6 @@
...
-| Custom resource           | Description                   |
-| ------------------- | ------------------------------------------------------------------------------------------------------------ |
-| [Function](./svls-01-function.md)     | Stores the Function's source code and dependencies on a cluster.  |
-
-These components use this CR:
-
-| Component           | Description                              |
-| ------------------- | ------------------------------------------------------------------------------------------------------------ |
-| Function Controller | Uses the GitRepository CR to locate the Function's source code and dependencies in a Git repository.
+The `gitrepositories.serverless.kyma-project.io` CustomResourceDefinition (CRD) was removed and definitions were moved to the `functions.serverless.kyma-project.io` CRD. For more information, see [Function](../00-custom-resources/svls-01-function.md)
```
Comment (dbadura) [inline docs/05-technical-reference/00-custom-resources/svls-02-gitrepository.md:7]:
```suggestion
The `gitrepositories.serverless.kyma-project.io` CustomResourceDefinition (CRD) was deprecated and definitions were moved to the `functions.serverless.kyma-project.io` in version `v1alpha2` CRD. For more information, see [Function](../00-custom-resources/svls-01-function.md)
```

File: docs/05-technical-reference/00-custom-resources/svls-02-gitrepository.md [hunk 21]
```diff
@@ -2,49 +2,6 @@
...
-| Custom resource           | Description                   |
-| ------------------- | ------------------------------------------------------------------------------------------------------------ |
-| [Function](./svls-01-function.md)     | Stores the Function's source code and dependencies on a cluster.  |
-
-These components use this CR:
-
-| Component           | Description                              |
-| ------------------- | ------------------------------------------------------------------------------------------------------------ |
-| Function Controller | Uses the GitRepository CR to locate the Function's source code and dependencies in a Git repository.
+The `gitrepositories.serverless.kyma-project.io` CustomResourceDefinition (CRD) was deprecated and definitions were moved to the `functions.serverless.kyma-project.io` in version `v1alpha2` CRD. For more information, see [Function](../00-custom-resources/svls-01-function.md)
```
Comment (grego952) [inline docs/05-technical-reference/00-custom-resources/svls-02-gitrepository.md:7]:
```suggestion
The `gitrepositories.serverless.kyma-project.io` CustomResourceDefinition (CRD) was deprecated and definitions were moved to the `functions.serverless.kyma-project.io` CRD in version `v1alpha2`. For more information, see [Function](../00-custom-resources/svls-01-function.md)
```

**Review by grego952 — APPROVED** [review]

**Pull request merged**